framework/src/seclink/seclink.c

```diff
@@ -124,7 +139,7 @@ int sl_init(sl_ctx *hnd)
 	handle->fd = fd;
 
 	struct seclink_req req = {.req_type.comm = NULL, 0};
-	SL_CALL(handle, SECLINKIOC_INIT, req);
+	SL_CALL2(handle, SECLINKIOC_INIT, req, SL_FREE_HANDLE(handle));
```

Comment:
As `SL_CALL2` can potentially return on error, shouldn't we also `close(fd)` inside the macro definition before returning? Please ignore if not necessary.